client/state/notices/middleware.js

```diff
@@ -122,6 +122,12 @@ export const onPublicizeConnectionCreate = ( dispatch, { connection } ) => dispa
 	} ) )
 );
 
+export const onPublicizeConnectionCreateFailure = ( dispatch, { error } ) => dispatch(
+	errorNotice( error.message || translate( 'An error occurred while connecting the account.', {
```

Comment:
If the messages in backend are understandable by normal folks, this is good to go 👍